lib/next_ls.ex

```diff
@@ -171,6 +173,66 @@ defmodule NextLS do
     {:reply, symbols, lsp}
   end
 
+  # TODO extract to a separate module
+  # TODO make this work for modules and structs
+  # TODO how to handle aliases?
+  # TODO can we do this in a single DB query?
+  def handle_request(%TextDocumentReferences{params: %{position: position, text_document: %{uri: uri}}} = request, lsp) do
+    import NextLS.DB.Query
+
+    file = URI.parse(uri).path
+    line = position.line + 1
+    col = position.character + 1
+
+    result =
+      dispatch(lsp.assigns.registry, :databases, fn databases ->
+        for {database_pid, _} <- databases do
+          query = ~Q"""
+          SELECT identifier, arity, type, module
+          FROM "references" as refs
+          WHERE refs.file = ?
+            AND refs.start_line <= ? AND refs.end_line >= ?
+            AND refs.start_column <= ? AND refs.end_column >= ?
+          ORDER BY refs.id ASC
+          LIMIT 1;
+          """
+
+          [[identifier, arity, type, module]] = DB.query(database_pid, query, [file, line, line, col, col])
+
+          query = ~Q"""
+          SELECT file, start_line, end_line, start_column, end_column
+          FROM "references" as refs
+          WHERE refs.identifier = ?
+            AND refs.arity = ?
+            AND refs.type = ?
+            AND refs.module = ?
```

Comment:
in the case that we are finding references for a reference (meaning, a usage of the function, not the definition), we'll want to exclude the reference we're on from the result set.

I think at least, would be valuable to see what other LSPs like gopls and rust analyzer do

Reply:
ElixirLS doesn't exclude the current reference but we can definitely do it. I'll definitely check what other LSPs do.